ci/pull_e2e_kind.groovy

```diff
@@ -0,0 +1,362 @@
+
+    // Git ref to checkout
+    def GIT_REF = params.GIT_REF
+    if (params.PR_ID != "") {
+        GIT_REF = "refs/remotes/origin/pull/${params.PR_ID}/head"
+    } else if (env.ghprbActualCommit) {
+        // for PR jobs triggered by ghprb plugin
+        GIT_REF = env.ghprbActualCommit
+    }
+
+    timeout (time: 2, unit: 'HOURS') {
+        // use fixed label, so we can reuse previous workers
+        // increase verison in pod label when we update pod template
+        def buildPodLabel = "tidb-operator-build-v1"
+        def resources = [
+            requests: [
+                cpu: "4",
+                memory: "4G"
+            ],
+            limits: [
+                cpu: "8",
+                memory: "32G"
+            ],
+        ]
+        podTemplate(
+            label: buildPodLabel,
+            yaml: buildPodYAML(resources: resources, any: true),
+            // We allow this pod to remain active for a while, later jobs can
+            // reuse cache in previous created nodes.
+            idleMinutes: 180,
+        ) {
+        node(buildPodLabel) {
+            container("main") {
+                dir("${PROJECT_DIR}") {
+
+                    stage('Checkout') {
+                        sh """
+                        echo "info: change ownerships for jenkins"
+                        # we run as root in our pods, this is required
+                        # otherwise jenkins agent will fail because of the lack of permission
+                        chown -R 1000:1000 .
+                        """
+
+                        // clean stale files because we may reuse previous created nodes
+                        deleteDir()
+
+                        checkout changelog: false, poll: false, scm: [
+                                $class: 'GitSCM',
+                                branches: [[name: "${GIT_REF}"]],
+                                userRemoteConfigs: [[
+                                    refspec: '+refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pull/*',
+                                    url: "${params.GIT_URL}",
+                                ]]
+                            ]
+
+                        GITHASH = sh(returnStdout: true, script: "git rev-parse HEAD").trim()
+                    }
+
+                    stage("Build and Test") {
+                        withCredentials([
+                            string(credentialsId: "tp-codecov-token", variable: 'CODECOV_TOKEN')
+                        ]) {
+                            sh """#!/bin/bash
+                            set -eu
+                            echo "info: building"
+                            make build e2e-build
+                            if [ "${GIT_REF}" == "master" ]; then
+                                echo "info: run unit tests and report coverage results for master branch"
+                                make test GOFLAGS='-race' GO_COVER=y
```

Comment:
do we need to run unit test in this job?

Reply:
in pull request jobs, no 
in master branch job, yes (reporting the test coverage for master branch)

Reply:
got it